// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.scp.cpio.private_key_client;

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

message ListPrivateKeysByIdsProtoRequest {
  repeated string key_ids = 1;
}

message ListPrivateKeysByIdsProtoResponse {
  repeated PrivateKeyProto private_keys = 1;
}

message PrivateKeyProto {
  // The id of the private key.
  string key_id = 1;
  // The value of the public key.
  string public_key = 2;
  // The value of the private key.
  string private_key = 3;
  // The expiration time of the private key.
  int64 expiration_time_in_ms = 4;
};